include/boost/beast/core/impl/dynamic_buffer.hpp

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright (c) 2020 Richard Hodges (hodges.r@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_CORE_IMPL_DYNAMIC_BUFFER_HPP
+#define BOOST_BEAST_CORE_IMPL_DYNAMIC_BUFFER_HPP
+
+#include <boost/beast/core/dynamic_buffer.hpp>
+#include <memory>
+
+namespace boost {
+namespace beast {
+
+template<class DynamicBuffer_v0>
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+dynamic_buffer_v0_proxy(
+    DynamicBuffer_v0& storage) noexcept
+: storage_(std::addressof(storage))
+{
+    boost::ignore_unused(storage_->prepare(0));
+}
+
+#if BOOST_BEAST_DOXYGEN
+
+template<class DynamicBuffer_v0>
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+dynamic_buffer_v0_proxy(
+    dynamic_buffer_v0_proxy const&) noexcept
+= default;
+
+template<class DynamicBuffer_v0>
+auto
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+operator=(
+    dynamic_buffer_v0_proxy const&) noexcept ->
+dynamic_buffer_v0_proxy&
+= default;
+
+template<class DynamicBuffer_v0>
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+~dynamic_buffer_v0_proxy() noexcept
+= default;
+
+#endif // BOOST_BEAST_DOXYGEN
+
+template<class DynamicBuffer_v0>
+auto
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+size() const ->
+std::size_t
+{
+    return storage_->size();
+}
+
+template<class DynamicBuffer_v0>
+auto
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+max_size() const ->
+std::size_t
+{
+    return storage_->max_size();
+}
+
+template<class DynamicBuffer_v0>
+auto
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+capacity() const ->
+std::size_t
+{
+    return storage_->capacity();
+}
+
+template<class DynamicBuffer_v0>
+auto
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+consume(std::size_t n) ->
+void
+{
+    storage_->consume(n);
+}
+
+template<class DynamicBuffer_v0>
+auto
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+data(std::size_t pos, std::size_t n) const ->
+const_buffers_type
+{
+    return detail::dynamic_buffer_v2_access::
+    data(static_cast<DynamicBuffer_v0 const&>(*storage_),
+         pos, n);
+}
+
+template<class DynamicBuffer_v0>
+auto
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+data(std::size_t pos, std::size_t n) ->
+mutable_buffers_type
+{
+    return detail::dynamic_buffer_v2_access::
+    data(*storage_, pos, n);
+}
+
+template<class DynamicBuffer_v0>
+auto
+dynamic_buffer_v0_proxy<DynamicBuffer_v0>::
+grow(std::size_t n) ->
+void
+{
+    storage_->commit(net::buffer_size(storage_->prepare(n)));
```

Comment:
Beast provides `buffer_bytes` for this purpose, it is preferable to `net::buffer_size`